packages/xstate-fsm/src/index.ts

```
@@ -186,8 +182,18 @@ export function createMachine<
 
           const isTargetless = target === undefined;
 
+          const nextStateValue = target ?? value;
+          const nextStateConfig = fsmConfig.states[nextStateValue];
```

q: why the code has been moved? this has previously been resolved after passing a guard

---

My thinking was that the result of the guard doesn't matter much if the `nextStateConfig` is invalid, so I figured I would throw the error early.

---

Should I change anything here?

---

no, it's fine